### PR TITLE
Add CREATE VIEW

### PR DESCRIPTION
**Pull request by jklukas**

### CODE REVIEW

**Comment by jklukas** [comment]
Tests are failing due to `HTTPError: 503 Server Error: Service Unavailable` when posting to bigcrunch.

**Comment by graingert** [comment]
I suspect this is because we're waiting longer than 30 seconds for Redshif to boot.

I'll add polling on the tests side

File: redshift_sqlalchemy/dialect.py [hunk 1]
```diff
@@ -1145,3 +1155,48 @@ def visit_delete_stmt(element, compiler, **kwargs):
...
+                        for column in element.columns]
+        self.column_names = [column.key for column in element.columns]
+        self.selectable = selectable
+
+
+@compiles(CreateView)
+def visit_create_view(create, compiler, **kw):
+    view = create.element
+    preparer = compiler.dialect.identifier_preparer
+    text = "\nCREATE VIEW %s " % preparer.format_table(view)
```
Comment (graingert) [inline redshift_sqlalchemy/dialect.py]:
```python
"""
CREATE VIEW {table}
{columns}
AS {query}
""".format(...)

**Comment by graingert** [comment]
Do other dialects have a CreateView construct?

**Comment by jklukas** [comment]
> Do other dialects have a CreateView construct?

Not that I'm aware of. There are comments in the code that discuss an API for views as a future feature of core SQLAlchemy, but nothing concrete yet. There's also a [CreateView recipe](https://bitbucket.org/zzzeek/sqlalchemy/wiki/UsageRecipes/Views) on the SQLAlchemy wiki.

I have an immediate use case for CreateView constructs, and it seems like something we can strip out or modify from this dialect once a view API in core SQLAlchemy gets settled.

**Comment by graingert** [comment]
@jklukas why not just publish https://bitbucket.org/zzzeek/sqlalchemy/wiki/UsageRecipes/Views verbatim as a package?

**Comment by jklukas** [comment]
> why not just publish https://bitbucket.org/zzzeek/sqlalchemy/wiki/UsageRecipes/Views verbatim as a new package?

That's not a bad idea. I don't have a comprehensive knowledge of `CREATE TABLE` across dialects, but Redshift's version doesn't look to have any features that aren't generally supported by other dialects.

I'll close this.

**Comment by jklukas** [comment]
Note that this package now exists at https://github.com/jklukas/sqlalchemy-views

**Pull request closed without merging**